listening-comp/backend/progress_tracker.py

```
@@ -0,0 +1,369 @@
+        
+        for record in records:
+            progress_id, transcript_id, questions_attempted, questions_correct, \
+                last_access, completion_time, jlpt_level, notes, \
+                source_url, metadata = record
+            
+            accuracy = round(questions_correct / questions_attempted * 100, 2) if questions_attempted > 0 else 0
+            
+            # Parse metadata JSON
+            try:
+                metadata_dict = json.loads(metadata) if metadata else {}
+            except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)